ballet/eng/misc.py

```diff
@@ -1,18 +1,43 @@
 import funcy
 import numpy as np
 import pandas as pd
+from scipy.special import boxcox1p
+from scipy.stats import skew
 
 from ballet.eng.base import BaseTransformer, SimpleFunctionTransformer
 from ballet.util import get_arr_desc
 
-__all__ = ['IdentityTransformer', 'ValueReplacer', 'NamedFramer']
+__all__ = ['IdentityTransformer', 'BoxCoxTransformer', 'ValueReplacer', 'NamedFramer']
 
 
 class IdentityTransformer(SimpleFunctionTransformer):
     def __init__(self):
         super().__init__(funcy.identity)
 
 
+class BoxCoxTransformer(BaseTransformer):
+    def __init__(self, threshold, lmbda=0):
+        super().__init__()
+        self.threshold = threshold
+        self.lmbda = lmbda
+
+    def fit(self, X, y=None, **fit_args):
+        self.features_to_transform_ = abs(skew(X)) > self.threshold
+        if (isinstance(X, pd.DataFrame)):
+            # Hack to get a mask over columns
+            self.features_to_transform_ = X.columns.putmask(~self.features_to_transform_, None) & X.columns
+        return self
+
+    def transform(self, X, **transform_args):
+        if isinstance(X, pd.DataFrame):
```

Comment:
We want to apply boxcox to only those features we have identified in `fit` and pass through the others unchanged. Seems to me that if `self.features_to_tranform_ == []`, then `X[[]]` is an empty matrix and nothing is returned. The expected behavior is all columns are returned unchanged. A test case should be written for this phenomenon and `transform` changed accordingly.